es6/timeline.js

```diff
@@ -69,6 +121,28 @@ class TimelineComponent extends React.Component {
 
     render() {
         const {queue, warped_time, debug} = this.props
+        const anim_list = []
+        let container_width = 0
+        for (let idx in queue){
```

Comment:
The `for... in` uses the index of the array, and we need the index so is the same that the other for:
`for (let idx in queue)`  == `for (let i=0; i<queue.length; i++)`

Reply:
Ah, I forgot `for in` exists, I try to avoid for-in since it misbehaves if the array has properties added that aren't numbers.  Can we change it to `(let idx of range(queue.length)` or `i=0;i<...`?